src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +17,176 @@ import {
+  //     const randomAvatar = randomAvatarUrl(user.id);
+  //     setPictureUrl(randomAvatar);
+  //
+  //     // TODO: this isn't very reacty.. is it even needed..?
+  //     // event.currentTarget.onerror = null;
+  //     // event.currentTarget.src = randomAvatar;
+  //   },
+  //   [user.id]
+  // );
+
+  const containerClasses = classNames(
+    "UserProfilePicture",
+    `UserProfilePicture--reaction-${reactionPosition}`
+  );
+
+  const avatarClasses = classNames(
+    "UserProfilePicture__avatar",
+    avatarClassName
+  );
 
+  const avatarStyles = useMemo(
+    () => ({
+      backgroundImage: `url(${pictureUrl})`,
+      ...avatarStyle,
+    }),
+    [avatarStyle, pictureUrl]
+  );
+
+  const userDisplayName = user.anonMode ? DEFAULT_PARTY_NAME : user.partyName;
+
+  return (
+    <div className={containerClasses} style={containerStyle}>
+      {/* TODO: extract this logic into a generic custom component + improve it? Or just remove it entirely? */}
+      {/* Hidden image, used to handle error if image is not loaded */}
+      {/*<img*/}
+      {/*  src={pictureUrl}*/}
+      {/*  onError={(event) => imageErrorHandler(event)}*/}
+      {/*  hidden*/}
+      {/*  style={{ display: "none" }}*/}
+      {/*  alt={user.anonMode ? DEFAULT_PARTY_NAME : user.partyName}*/}
+      {/*/>*/}
+
+      {/* TODO: can we use src/components/atoms/UserAvatar/UserAvatar.tsx here? Should we? */}
+      <div
+        role="img"
+        aria-label={`${userDisplayName}'s avatar`}
+        className={avatarClasses}
+        style={avatarStyles}
+        onClick={openProfileModal}
+      />
+
+      {/* TODO: refactor this to be more performant and potentially memo it if required */}
+      <div className="UserProfilePicture__reaction-container">
         {Reactions.map(
           (reaction, index) =>
             reactions.find(
               (r) => r.created_by === user.id && r.reaction === reaction.type
             ) && (
-              <div key={index} className="reaction-container">
-                <S.Reaction
+              // TODO: don't use index as a key
+              // TODO: should we extract reactions into their own sub-components?
+              // TODO: is reaction-container even defined in our styles here..?
+              <React.Fragment key={index}>
+                <div
+                  className="UserProfilePicture__reaction"
                   role="img"
                   aria-label={reaction.ariaLabel}
-                  className={reaction.name}
-                  reactionPosition={reactionPosition}
                 >
                   {reaction.text}
-                </S.Reaction>
+                </div>
 
+                {/* TODO: replace this with useSound or similar */}
                 {!muteReactions && !isAudioEffectDisabled && (
                   <audio autoPlay loop>
                     <source src={reaction.audioPath} />
                   </audio>
                 )}
```

Comment:
[TODO] replace this with `useSound` or similar

Reply:
OOS, marked this as `@debt` https://github.com/sparkletown/sparkle/pull/1323/commits/af6335be218415776a2481a7ad2a1aebb7d343b0